File: app/src/main/java/com/vladimir_x/easyenglishlearn/model/Answer.java
package com.vladimir_x.easyenglishlearn.model;

public class Answer {
    private Word mQuestion;
    private String mAnswer;
    private boolean mTranslationDirection;

    public Answer(Word question, StringBuilder answerBuilder, boolean translationDirection) {
        mQuestion = question;
        mAnswer = answerBuilder.toString();
        mTranslationDirection = translationDirection;
    }

    public boolean isCorrect() {
        return mTranslationDirection
                ? mQuestion.getTranslation().equalsIgnoreCase(mAnswer)
                : mQuestion.getLexeme().equalsIgnoreCase(mAnswer);
    }

}

File: app/src/main/java/com/vladimir_x/easyenglishlearn/model/Answer.kt
package com.vladimir_x.easyenglishlearn.model

class Answer(
    private val question: Word?,
    answer: CharSequence,
    private val translationDirection: Boolean
) {
    private val answer: String = answer.toString()
    val isCorrect: Boolean
        get() = if (translationDirection) {
            question?.translation.equals(answer, ignoreCase = true)
        } else {
            question?.lexeme.equals(answer, ignoreCase = true)
        }

}